gap/Magic.gd

```diff
@@ -56,15 +60,27 @@
 #!
 #! <Mark><A>package</A></Mark>
 #! <Item>
-#!     This is either the name of package, or an <C>IsDirectory</C> object.
-#!     In the former case, &AutoDoc; uses the metadata of the first package
-#!     with that name known to &GAP;. In the latter case, it checks whether
-#!     the given directory contains a <F>PackageInfo.g</F> file, and extracts
-#!     all needed metadata from that. This is for example useful if you have
-#!     multiple versions of the package around and want to make sure the
+#!     The purpose of this parameter is twofold: to determine the package
+#!     directory in which &AutoDoc; will operate, and to find the metadata
+#!     concerning the package being documented. The parameter is either a
+#!     string, an <C>IsDirectory</C> object, or omitted. 
+#!     In the first case, &AutoDoc; interprets the string as the name of a
+#!     package, uses the metadata of the first package known to &GAP;
+#!     with that name, and uses the <C>InstallationPath</C> specified in that
+#!     metadata as the package directory. The other cases directly specify the
+#!     package directory, either given by the <C>IsDirectory</C> object or as
+#!     the <C>DirectoryCurrent()</C> if the <A>package</A> parameter is
+#!     omitted. In these cases, the package directory must contain a
+#!     <F>PackageInfo.g</F> file, and &AutoDoc; extracts all needed metadata
+#!     from that. The <C>IsDirectory</C> form is for example useful if you
+#!     have multiple versions of the package around and want to make sure the
 #!     documentation of the correct version is built.
 #!     <P/>
-#!     If this argument is omitted, &AutoDoc; uses the <C>DirectoryCurrent()</C>.
+#!     Note that when using <C>AutoDocWorksheet</C>
+#!     (see <Ref Sect='Chapter_AutoDoc_worksheets_Section_Worksheets' />), there
+#!     is no parameter corresponding to <A>package</A> and so the "package
+#!     directory" is always the <C>DirectoryCurrent()</C> and there is no
+#!     metadata.
```

Comment:
@sebasguts actually, why is that? Why can't one specify a target directory for worksheets?

Reply:
Yeah, this is somehow broken by design I would say. The `AutoDocWorksheet` command sets the package name to `AutoDocWorksheet`...

Reply:
I would say this should probably be resolved by passing an option `Worksheet := true` to the `AutoDoc` command